lib/app/utils/startup_tasks.rb

```diff
@@ -44,10 +46,27 @@ def check_validator_availability
         end
       end
 
+      def load_profiles_in_validator(module_metadata)
+        ig_files(module_metadata[:resource_path]).each do |full_file_path|
+          next if File.directory?(full_file_path)
+
+          begin
+            contents = File.read(full_file_path)
+            next unless JSON.parse(contents)['resourceType'] == 'StructureDefinition'
+
+            RestClient.post("#{validator_url}/profiles", contents)
+          rescue StandardError => e
+            Inferno.logger.error "Unable to post profile '#{File.basename(full_file_path)}' to validator"
```

Comment:
Did you mean that it says anything that isn't JSON is a profile? Because I have a guard statement that ensures only `StructureDefinition`s are posted, but I could see how the error message could be misleading if `JSON.parse` raises an error. I've added a `rescue JSON::ParserError` clause to fix this (haven't pushed yet)

Reply:
Yes, I was thinking about parsing errors in particular.